src/FINDOLOGIC/FindologicApi.php

```diff
@@ -218,13 +218,14 @@ private function checkResponseIsValid($requestType, $responseBody, $statusCode)
     {
         $isAlivetestRequest = $requestType === RequestType::ALIVETEST_REQUEST;
         $responseBodyIsAlive = $responseBody == self::SERVICE_ALIVE_BODY;
-        $isAlivetestAndBodyIsOk = $isAlivetestRequest && $responseBodyIsAlive;
         $httpCodeIsOk = $statusCode === self::STATUS_OK;
 
-        // If it is an alivetest, the 'alive' body needs to be set. If it is not an alivetest, then we do not
-        // care about the body. The http code always needs to be 200 OK.
-        if (!(($isAlivetestAndBodyIsOk && $httpCodeIsOk) || !$isAlivetestRequest && $httpCodeIsOk)) {
-            throw new ServiceNotAliveException($responseBody);
+        if ($isAlivetestRequest) {
```

Comment:
Maybe using something like


```php
switch ($requestType) {
  RequestType::ALIVETEST_REQUEST:
    …
  RequestType:…
```

could make this more readable.

Reply:
I changed this completely anyway :).